### PR TITLE
Add support for multiple paths in rz-find

### LINKED ISSUE #5253: Support multiple paths (files and directories) in `rz-find`
`rz-find` should support `rz-find ... [path1] [path2] ...` multipath search - either multiple
files or multiple directories, just like `find` tool

### PR DESCRIPTION
**Pull request by IndAlok**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/dev/CONTRIBUTING.md) to this repository.
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#code-style).
- [ ] I've documented every `RZ_API` function and struct this PR changes. *(No `RZ_API` interfaces were changed.)*
- [x] I've added tests that prove my changes are effective (required for changes to `RZ_API`). *(Tests added/updated for `rz-find`; no `RZ_API` changes.)*
- [ ] I've updated the [Rizin book](https://github.com/rizinorg/book) with the relevant information (if needed). *(Not needed for this change.)*
- [ ] I've used AI tools to generate fully or partially these code changes and I'm sure the changes are not copyrighted by somebody else.

---

## Detailed description

This PR enhances **`rz-find`** to support **multiple input paths** (multiple files and/or directories) in a single invocation, similar to how coreutils `find` accepts multiple starting points.

### What this changes
- Accept and process `rz-find ... [path1] [path2] ...` where each path may be a file or a directory.
- Continue scanning subsequent paths even if one path fails to open or scan (error is reported, but the tool keeps going).
- Return an aggregated exit status: success only if all provided paths were processed successfully.
- Make output consistent across multi-path use:
  - Non-JSON output prints per-file headers when needed (so results remain attributable).
  - Verbose “Scanning:” diagnostics are emitted in a way that does not corrupt JSON output. (Earlier was getting an extra "[" :sad )

### Implementation notes
- Adjusted the main iteration logic to properly handle multiple independent paths (files/dirs), including edge cases like empty paths and partial failures.
- Fixed JSON framing/comma handling so multi-path output remains valid JSON.
- Updated/added tests and updated `rz-find` documentation accordingly.

---

## Test plan

Commands (Windows / PowerShell example, on which it was tested):

1. Configure + build:
````powershell
meson setup build --buildtype=debug
ninja -C build
````

2. Run `rz-find` regression tests:
````powershell
cd test
..\build\binrz\rz-test\rz-test.exe -j 1 db/tools/rz_find
````

3. Spot-check multi-path behavior manually (examples):
````powershell
..\build\binrz\rz-find\rz-find.exe -s README bins\arm\README bins\arm
..\build\binrz\rz-find\rz-find.exe -j -s README bins\arm\README bins\arm\README
..\build\binrz\rz-find\rz-find.exe -V -j -s README bins\arm\README bins\arm\README
````

(If JSON validation is desired, parse stdout with a JSON parser; ensure verbose output does not break JSON.)

---

## Closing issues

closes #5253

### CODE REVIEW

File: librz/main/rz-find.c [hunk 1]
```diff
@@ -175,19 +257,27 @@ static void print_bin_string(RzBinFile *bf, RzBinString *string, PJ *pj) {
...
-		pj_ks(pj, "string", string->string);
-		pj_end(pj);
+		// Escape all string fields for JSON safety
+		char *escaped_section = rz_str_escape_utf8_for_json(section_name, -1);
+		char *escaped_type = rz_str_escape_utf8_for_json(type_string, -1);
+		char *escaped_string = rz_str_escape_utf8_for_json(string->string, -1);
+		printf("%s{\"vaddr\":%" PFMT64u ",\"paddr\":%" PFMT64u ",\"size\":%" PFMT64u ",\"length\":%" PFMT64u ",\"section\":\"%s\",\"type\":\"%s\",\"string\":\"%s\"}",
+			ro->comma ? ro->comma : "",
+			string->vaddr, string->paddr, (ut64)string->size, (ut64)string->length,
+			escaped_section ? escaped_section : "",
```
Comment (wargio) [inline librz/main/rz-find.c]:
```suggestion
			rz_str_get(escaped_section),
```

File: librz/main/rz-find.c [hunk 2]
```diff
@@ -239,16 +329,25 @@ static int rzfind_open_file(RzfindOptions *ro, const char *file, const ut8 *data
...
+		char *rizin_path = rzfind_resolve_rizin_path();
+		const char *args[] = {
+			"-e",
+			"search.show=false",
+			"-e",
+			"search.maxhits=1",
+			"-nqcpm",
+			file,
+		};
+		rzfind_run_subprocess(rizin_path, args, RZ_ARRAY_SIZE(args));
```
Comment (wargio) [inline librz/main/rz-find.c]: while this is ok, i would strongly prefer to call the C api. i guess for now it is still ok.

**Review by wargio — CHANGES_REQUESTED** [review]
definitely much better.

File: librz/main/main.c [hunk 3]
```diff
@@ -20,7 +20,7 @@ static MainEntry main_prog[] = {
 	{ "rz-ax", rz_main_rz_ax },
 	{ "rz-bin", rz_main_rz_bin },
 	{ "rz-diff", rz_main_rz_diff },
-	{ "rz-find", rz_main_rz_ax },
+	{ "rz-find", rz_main_rz_find },
```
Comment (notxvilka) [inline librz/main/main.c:23]: Hah, nice catch!

**Review by notxvilka — COMMENTED** [review]
Great improvement! Exactly what I had in mind for this feature. Please address existing feedback then we can merge

File: librz/main/rz-find.c [hunk 4]
```diff
@@ -65,6 +66,80 @@ typedef struct {
...
+	free(self_path);
+	if (!self_dir) {
+		return rz_str_dup("rizin");
+	}
+
+#if __WINDOWS__
+	const char *rizin_exe = "rizin.exe";
+#else
+	const char *rizin_exe = "rizin";
+#endif
```
Comment (wargio) [inline librz/main/rz-find.c]: shouldn't this `rizin_exe` be used in the `rz_str_dup("rizin");` above ?

File: librz/main/rz-find.c [hunk 5]
```diff
@@ -175,19 +257,27 @@ static void print_bin_string(RzBinFile *bf, RzBinString *string, PJ *pj) {
...
+		// Escape all string fields for JSON safety
+		char *escaped_section = rz_str_escape_utf8_for_json(section_name, -1);
+		char *escaped_type = rz_str_escape_utf8_for_json(type_string, -1);
+		char *escaped_string = rz_str_escape_utf8_for_json(string->string, -1);
+		printf("%s{\"vaddr\":%" PFMT64u ",\"paddr\":%" PFMT64u ",\"size\":%" PFMT64u ",\"length\":%" PFMT64u ",\"section\":\"%s\",\"type\":\"%s\",\"string\":\"%s\"}",
+			ro->comma ? ro->comma : "",
+			string->vaddr, string->paddr, (ut64)string->size, (ut64)string->length,
+			rz_str_get(escaped_section),
+			escaped_type ? escaped_type : "",
+			escaped_string ? escaped_string : "");
```
Comment (wargio) [inline librz/main/rz-find.c]: `rz_str_get(xx)` is the equivalent of `xx ? xx : ""` so you should change all of them

File: librz/main/rz-find.c [hunk 6]
```diff
@@ -138,7 +139,7 @@ static int hit(RzSearchKeyword *kw, void *user, ut64 addr) {
 	if (ro->json) {
 		const char *type = "string";
 		printf("%s{\"offset\":%" PFMT64d ",\"type\":\"%s\",\"data\":\"%s\"}",
-			ro->comma, addr, type, str);
+			ro->comma ? ro->comma : "", addr, type, str);
```
Comment (notxvilka) [inline librz/main/rz-find.c]: `rz_str_get()`

File: librz/main/rz-find.c [hunk 7]
```diff
@@ -166,29 +167,44 @@ static int hit(RzSearchKeyword *kw, void *user, ut64 addr) {
...
-		pj_ks(pj, "section", section_name);
-		pj_ks(pj, "type", type_string);
-		pj_ks(pj, "string", string->string);
-		pj_end(pj);
+		// Escape all string fields for JSON safety
+		char *escaped_section = rz_str_escape_utf8_for_json(section_name, -1);
+		char *escaped_type = rz_str_escape_utf8_for_json(rz_str_get(type_string), -1);
+		char *escaped_string = rz_str_escape_utf8_for_json(rz_str_get(string->string), -1);
+		printf("%s{\"vaddr\":%" PFMT64u ",\"paddr\":%" PFMT64u ",\"size\":%" PFMT64u ",\"length\":%" PFMT64u ",\"section\":\"%s\",\"type\":\"%s\",\"string\":\"%s\"}",
+			ro->comma ? ro->comma : "",
```
Comment (notxvilka) [inline librz/main/rz-find.c]: `rz_str_get()`

**Review by notxvilka — CHANGES_REQUESTED** [review]

File: librz/main/rz-find.c (hunk 4 above)
Comment (IndAlok) [reply librz/main/rz-find.c]: yep shoudl've been, not runnable as a binary directly. but I rather considered the suggestion of using c api as you suggested, in recent commit(s).

File: librz/main/rz-find.c (hunk 5 above)
Comment (IndAlok) [reply librz/main/rz-find.c]: sure, was marked as draft until I replaced the remaining ones for that...

File: librz/main/rz-find.c (hunk 7 above)
Comment (IndAlok) [reply librz/main/rz-find.c]: my bad, done.

**Review by notxvilka — APPROVED** [review]

File: test/db/tools/rz_find [hunk 8]
```diff
@@ -284,34 +286,113 @@ RUN
...
+NAME=rz-find verbose with JSON outputs scanning to stderr
+FILE==
+CMDS=!rz-find -V -j -s 250382 bins/elf/ioli/crackme0x00
+EXPECT=<<EOF
+[{"offset":1423,"type":"string","data":"250382"}]
+EOF
+EXPECT_ERR=<<EOF
+Scanning: bins/elf/ioli/crackme0x00
+EOF
+RUN
```
Comment (wargio) [inline test/db/tools/rz_find]: i know this sounds weird, but maybe is better to join these test so we don't spawn 2 processes each time, but only one.

Comment (wargio) [reply test/db/tools/rz_find]: you can split the results via `echo ---`

Comment (IndAlok) [reply test/db/tools/rz_find]: seems creative, thanks for the suggestion.
